### PR TITLE
[Synthetics] Support proxy environment variables

### PR DESCRIPTION
**Pull request by lefebvree**
### What and why?

Follow-up of https://github.com/DataDog/datadog-ci/pull/342 to add proper proxy environment variables support across `synthetics` command.

Add `HTTP_PROXY`/`HTTPS_PROXY`/`NO_PROXY` environment variables support for API calls and tunnel connections.

### How?

- Remove `disableEnvironmentVariables` API option for the `synthetics` command
- Always set tunnel Websocket agent (similar to API) to benefit from `ProxyAgent` env vars support
- Add a mocked Synthetics endpoints and tunnel Websocket proxy for proxy tests

### Review checklist

- [x] Feature or bugfix MUST have appropriate tests (unit, integration)

### CODE REVIEW

**Comment by m-rousse** [comment]
Other commands are using `ProxyAgent` to get proxy configuration from environment variables: https://github.com/DataDog/datadog-ci/pull/342
Is this something we could use here to have the same solution in all commands?

File: src/commands/synthetics/__tests__/tunnel.test.ts [hunk 1]
```diff
@@ -1,8 +1,11 @@
+// tslint:disable: no-string-literal
+
 import {PassThrough} from 'stream'
 
 import {mocked} from 'ts-jest/utils'
 
-import {ProxyConfiguration} from '../../../helpers/utils'
+import * as ciUtils from '../../../helpers/utils'
```
Comment (m-rousse) [inline src/commands/synthetics/__tests__/tunnel.test.ts:7]: Why this change?

**Review by m-rousse — APPROVED** [review]
This LGTM, congrats for the tests + thanks for the update to use proxy-agent.

File: src/helpers/utils.ts [hunk 2]
```diff
@@ -134,7 +134,7 @@ export const getRequestBuilder = (options: RequestOptions) => {
   return (args: AxiosRequestConfig) => axios.create(baseConfiguration)(overrideArgs(args))
 }
 
-const getProxyAgent = (proxyOpts?: ProxyConfiguration, disableProxyFromEnvVar?: boolean) => {
+export const getProxyAgent = (proxyOpts?: ProxyConfiguration, disableProxyFromEnvVar?: boolean) => {
```
[the remark this answers is not in the document]
Comment (lefebvree) [reply src/helpers/utils.ts:137]: Ok I was wondering if that was left only for us or for apps that would interface with datadog-ci

File: src/commands/synthetics/__tests__/tunnel.test.ts (hunk 1 above)
Comment (lefebvree) [reply src/commands/synthetics/__tests__/tunnel.test.ts:7]: Oops leftover
